config/configopaque/opaque.go

```diff
@@ -15,16 +15,19 @@
 package configopaque // import "go.opentelemetry.io/collector/config/configopaque"
 
 import (
-	"bytes"
 	"encoding"
 )
 
 // String alias that is marshaled in an opaque way.
 type String string
 
+const maskedString = "[REDACTED]"
+
+var maskedBytes = []byte(maskedString)
+
 var _ encoding.TextMarshaler = String("")
 
-// MarshalText marshals into a string of '*' of length equal to the opaque string.
+// MarshalText marshals the string as `[REDACTED]`.
 func (s String) MarshalText() ([]byte, error) {
-	return bytes.Repeat([]byte("*"), len(s)), nil
+	return maskedBytes, nil
```

Comment:
Not sure if the contract of this func is that the caller cannot change the return type. I would suggest:
```suggestion
	return []byte(maskedString), nil
```

Reply:
Is there no perf gain in caching the byte array as a var?

Reply:
May be, but it is incorrect since if caller does:

```
t := str.MarshalText()
t[2] = 0xFF
```

The result for future calls will be affected.

Reply:
Good point, will change.